init/recovery.go

```diff
@@ -0,0 +1,93 @@
+package init
+
+import (
+	log "github.com/Sirupsen/logrus"
+	composeConfig "github.com/docker/libcompose/config"
+	"github.com/docker/libcompose/yaml"
+	"github.com/rancher/os/compose"
+	"github.com/rancher/os/config"
+	"github.com/rancher/os/netconf"
+)
+
+var (
+	recoveryDockerService = composeConfig.ServiceConfigV1{
+		Image: config.OsBase,
+		Command: yaml.Command{
+			"ros",
+			"recovery-init",
+		},
+		Labels: map[string]string{
+			config.DetachLabel: "false",
+			config.ScopeLabel:  "system",
+		},
+		LogDriver:  "json-file",
+		Net:        "host",
+		Uts:        "host",
+		Pid:        "host",
+		Ipc:        "host",
+		Privileged: true,
+		Volumes: []string{
+			"/dev:/host/dev",
+			"/etc/ssl/certs/ca-certificates.crt:/etc/ssl/certs/ca-certificates.crt.rancher",
+			"/lib/modules:/lib/modules",
+			"/lib/firmware:/lib/firmware",
+			"/usr/bin/ros:/usr/bin/ros:ro",
+			"/usr/bin/ros:/usr/bin/cloud-init-save",
+			"/usr/bin/ros:/usr/bin/respawn:ro",
+			"/usr/share/ros:/usr/share/ros:ro",
+			"/var/lib/rancher:/var/lib/rancher",
+			"/var/lib/rancher/conf:/var/lib/rancher/conf",
```

Comment:
I wonder if we can add a command to exit the recovery console and continue booting